tests/unit/packet/PublishTest.php

```diff
@@ -225,65 +244,67 @@ public function testParseWithQos($qos, $bit)
     {
         $input =
             chr($bit) .
-            chr(4) .
+//            chr(4) .
```

Comment:
Hi. So, originally, this piece of code was in `parse()`...

```
        $packet->setMessage(
            substr(
                $rawInput,
                4 + strlen($topic)
            )
        );
```
or if I completely remove it...
```
//        $packet->setMessage(
//            substr(
//                $rawInput,
//                4 + strlen($topic)
//            )
//        );
```
the expected message in `testParseWithDup()` equals:
```
            chr(56) .
            chr(4) .
            chr(0) .
            chr(0) .
            chr(0) .
            chr(10);
```

I then changed the code to:
```
        $packet->addRawToPayLoad(
            substr(
                $rawInput,
                4 + strlen($topic)
            )
        );
```
or
```
        $packet->payload = substr(
            $rawInput,
            4 + strlen($topic)
        );
```
the expected message becomes:
```
            chr(56) .
            chr(2) .
            chr(0) .
            chr(0);
```
I can't really explain why that is. Do you understand it?